File: app/src/main/java/eu/darken/sdmse/common/forensics/csi/dalvik/tools/DirNameCheck.java
//package eu.darken.sdmse.common.forensics.csi.dalvik;
//
//import eu.thedarken.sdm.tools.forensics.Owner;
//import eu.thedarken.sdm.tools.forensics.OwnerInfo;
//import eu.thedarken.sdm.tools.forensics.csi.CSICheck;
//import eu.thedarken.sdm.tools.forensics.csi.CSIProcessor;
//import eu.thedarken.sdm.tools.io.FileOpsHelper;
//
//
//public class DirNameCheck extends CSICheck {
//
//    public DirNameCheck(CSIProcessor csiProcessor) {
//        super(csiProcessor);
//    }
//
//    @Override
//    public boolean check(OwnerInfo ownerInfo) {
//        String potPkg = FileOpsHelper.getFirstDirName(ownerInfo.getLocationInfo().getPrefixFreePath());
//        if (isInstalled(potPkg)) {
//            ownerInfo.addOwner(new Owner(potPkg));
//            return true;
//        }
//        return false;
//    }
//}

File: app/src/main/java/eu/darken/sdmse/common/forensics/csi/dalvik/tools/DirNameCheck.kt
package eu.darken.sdmse.common.forensics.csi.dalvik.tools

import dagger.Reusable
import eu.darken.sdmse.common.forensics.AreaInfo
import eu.darken.sdmse.common.forensics.Owner
import eu.darken.sdmse.common.forensics.csi.dalvik.DalvikCheck
import eu.darken.sdmse.common.pkgs.PkgRepo
import eu.darken.sdmse.common.pkgs.isInstalled
import eu.darken.sdmse.common.pkgs.toPkgId
import javax.inject.Inject

@Reusable
class DirNameCheck @Inject constructor(
    private val pkgRepo: PkgRepo,
) : DalvikCheck {

    suspend fun process(areaInfo: AreaInfo): DalvikCheck.Result {
        val potPkg = areaInfo.prefixFreeSegments.first().toPkgId()
        return if (pkgRepo.isInstalled(potPkg, areaInfo.userHandle)) {
            DalvikCheck.Result(setOf(Owner(potPkg, areaInfo.userHandle)))
        } else {
            return DalvikCheck.Result()
        }
    }
}